moveit_kinematics/test/test_kinematics_plugin.cpp

```diff
@@ -412,353 +390,358 @@ TEST_F(KinematicsTest, randomWalkIK)
+      success += solutions.empty() ? 0 : 1;
+    else
+      continue;
+
+    std::vector<geometry_msgs::msg::Pose> reached_poses;
+    for (const auto& s : solutions)
+    {
+      kinematics_solver_->getPositionFK(fk_names, s, reached_poses);
+      EXPECT_NEAR_POSES(poses, reached_poses, tolerance_);
+    }
+  }
+
+  if (num_ik_cb_tests_ > 0)
+  {
+    RCLCPP_INFO_STREAM(LOGGER, "Success Rate: " << (double)success / num_ik_multiple_tests_);
+  }
+  EXPECT_GE(success, EXPECTED_SUCCESS_RATE * num_ik_multiple_tests_);
+}
+
+// validate that getPositionIK() retrieves closest solution to seed
+TEST_F(KinematicsTest, getNearestIKSolution)
+{
+  std::vector<std::vector<double>> solutions;
+  kinematics::KinematicsQueryOptions options;
+  kinematics::KinematicsResult result;
+
+  std::vector<double> seed, fk_values, solution;
+  moveit_msgs::msg::MoveItErrorCodes error_code;
+  const std::vector<std::string>& fk_names = kinematics_solver_->getTipFrames();
+  moveit::core::RobotState robot_state(robot_model_);
+  robot_state.setToDefaultValues();
+
+  for (unsigned int i = 0; i < num_nearest_ik_tests_; ++i)
+  {
+    robot_state.setToRandomPositions(jmg_, this->rng_);
+    robot_state.copyJointGroupPositions(jmg_, fk_values);
+    std::vector<geometry_msgs::msg::Pose> poses;
+    ASSERT_TRUE(kinematics_solver_->getPositionFK(fk_names, fk_values, poses));
+
+    // sample seed vector
+    robot_state.setToRandomPositions(jmg_, this->rng_);
+    robot_state.copyJointGroupPositions(jmg_, seed);
+
+    // getPositionIK for single solution
+    kinematics_solver_->getPositionIK(poses[0], seed, solution, error_code);
+
+    // check if getPositionIK call for single solution returns solution
+    if (error_code.val != error_code.SUCCESS)
+      continue;
+
+    const Eigen::Map<const Eigen::VectorXd> seed_eigen(seed.data(), seed.size());
+    double error_get_ik =
+        (Eigen::Map<const Eigen::VectorXd>(solution.data(), solution.size()) - seed_eigen).array().abs().sum();
+
+    // getPositionIK for multiple solutions
+    solutions.clear();
+    kinematics_solver_->getPositionIK(poses, seed, solutions, result, options);
+
+    // check if getPositionIK call for multiple solutions returns solution
+    EXPECT_EQ(result.kinematic_error, kinematics::KinematicErrors::OK)
+        << "Multiple solution call failed, while single solution call succeeded";
+    if (result.kinematic_error != kinematics::KinematicErrors::OK)
+      continue;
+
+    double smallest_error_multiple_ik = std::numeric_limits<double>::max();
+    for (const auto& s : solutions)
+    {
+      double error_multiple_ik =
+          (Eigen::Map<const Eigen::VectorXd>(s.data(), s.size()) - seed_eigen).array().abs().sum();
+      if (error_multiple_ik <= smallest_error_multiple_ik)
+        smallest_error_multiple_ik = error_multiple_ik;
+    }
+    EXPECT_NEAR(smallest_error_multiple_ik, error_get_ik, tolerance_);
+  }
+}
 
 int main(int argc, char** argv)
 {
```

Comment:
Maybe add a  rclcpp::shutdown() before returning the result to properly shutdown the context

Reply:
I think the shutdown is called automatically in the end even if you don't write it in the context of tests.